keytree/keytree.md

```diff
@@ -42,18 +42,20 @@ Stands for _extended private key_: consists of a secret [scalar](#scalar) and a
 ```rust
 struct Xprv {
   scalar: Scalar,
-  dk: [u8; 32]
+  dk: [u8; 32],
+  transcript: Transcript,
 }
 ```
 
 ### Xpub
 
-Stands for _extended public key_: consists of a [point](#point) and a [derivation key](#derivation-key).
+Stands for _extended public key_: consists of a [point](#point), a [derivation key](#derivation-key), and a transcript.
```

Comment:
i'd remove the `transcript` field from the spec - that's the implementation detail. From the user's perspective, there's only a key and dk, and derivation rules from them. We simply precompute and reuse transcript for efficiency.

Reply:
Hmm. I see where you're going with that, but doesn't this spec include implementation details? E.g. steps 1-6 for "derive an intermediate key" seem like an implementation detail.

Reply:
_trying to articulate it better..._

The spec is focused on the _protocol_, while where we put the transcript is an implementation detail. We can instead store compressed pubkey (along the decompressed one) and recompute transcript anew on each derivation, that'd be almost as efficient. Or we could compress it on the fly and make things slower. Or we could store the compressed only, and avoid redundancy, but then derivation API would be failable which isn't nice. The protocol does not change regardless of those choices, and i think it's better to keep those choices out of the spec to allow other implementors focus on what actually matters for users, instead of micromanaging the implementation.